manifests/charts/istio-control/istio-discovery/files/injection-template.yaml

```diff
@@ -467,6 +471,12 @@ spec:
     configMap:
       name: istio-ca-root-cert
   {{- end }}
+  {{- if eq .Values.global.pilotCertProvider "kubernetes" }}
+  - name: kube-ca-cert
+    configMap:
+      name: kube-root-ca.crt
+      optional: true
```

Comment:
is it optional?

Reply:
If automount is not disabled, then the default mount path will also be present and the code now checks for that path if the file doesnt exist in the istio mount. So I considered this optional (even though both read from the same configmap).
We can also remove the optional and thus remove the extra if conditions in the istio agent code and just replace the k8sCAPath to this new path.